### ISSUE TITLE
Default cache-path not specific enough

### ISSUE THREAD

**Issue by UnitedMarsupials** [andreas-kupries/critcl#55]
critcl tries to use a different subdirectory under `~/.critcl` for each `$OS-$ARCH` combination -- in anticipation that the same home directory may be shared by the user on multiple computers.

Unfortunately, this is not enough -- and the operating system's release number should be part of it too. Because binaries built, for example, on FreeBSD-8 would be different from those built on FreeBSD-10, even if the architecture is the same. Same, likely, goes for SunOS, Linux-distros, et cætera.

**Comment by andreas-kupries** [comment]
Looking over the code the `targetplatform` information likely comes from the `buildplatform` information, which is initialized to the result of `platform::generic`.
This is the reduced platform information which has things like OS differences removed.
The full identification requires `platform::identify`.

Notes:
  1. The `platform` package is part of the Tcl core distribution. Definitely in 8.6+. Cannot recall when in the 8.5.x series, i.e. which patchlevel, it was added. Easiest to check is to run `package require platform` and see if that returns an error or not.
  1. As an aside, `critcl` comes with a `critcl::platform` package for when Tcl does not provide it (8.4.x for example).
  1. I do not recall what the output of `platform::identify` would be for FreeBSD.
    I suspect it does not take the FreeBSD differences into account there yet (because we had no FreeBSD machines around when it was written).
    It does account for Solaris kernel differences, and Linux glibc version differences..
  1. I would recommend to run `platform::identify` on your machines. Please provide patches to the Tcl core if the data coming back from the command is wrong.

**Comment by andreas-kupries** [comment]
Mostly addressed in commit 22cb4f3f0fe8b75ea68a299ef200e72c8c854e87.
I still need information about the output of `platform::identify` on "FreeBSD" to say if that needs an update or not.

**Comment by andreas-kupries** [comment]
Closing. Aforementioned commit has the switch `generic` to `identify`.
The info about FreeBSD has to go through the core's platform package.
The relevant tracker is at https://core.tcl.tk/tcl/tktnew

**Comment by UnitedMarsupials** [comment]
```
% package require critcl
3.1.15
% package require critcl::platform
1.0.11
% platform::identify
freebsd-amd64
```

Back in 2006 I created an account (`aku`) for you on one of my machines. The account is still active -- you are most welcome to use it for testing any of your software on FreeBSD.

**Comment by andreas-kupries** [comment]
Huh. Can you give me the name/IP of that machine ?
Do I need a password as well ?

**Comment by andreas-kupries** [comment]
Ok. This issue should now be fully addressed, with commits 136cab3c7328ceb5b750b991a133105d926adfac and 4152c8edca1e5266e96e94174b3cab59150be68d.

The internal package `critcl::platform` now incorporates the FreeBSD ABI version, and this package is preferred while the official package in the core does not do the same.

Note, while I tested on your machine, I had to make some local changes to ensure that my test code used the new `critcl::platform` package (@/home/aku/opt) instead of the older one from the global `critcl` installation (@/opt). As such I would like to ask you to do your own check as well, with a properly installed `critcl`, and your own examples.
